pkg/appgateserver/server.go

```diff
@@ -143,31 +178,19 @@ func (app *appGateServer) ServeHTTP(writer http.ResponseWriter, request *http.Re
 	// Extract the serviceId from the request path.
 	path := request.URL.Path
 	serviceId := strings.Split(path, "/")[1]
-	appAddress := request.URL.Query().Get("senderAddr")
-
-	// ensure the app address is present
-	if appAddress == "" && app.appAddress == "" {
-		app.replyWithError(
-			writer,
-			sdkerrors.Wrapf(ErrInvalidRequestURL, "missing sender address query parameter: got %s", request.URL.String()),
-		)
-		log.Print("ERROR: no application address provided")
-		return
-	} else if appAddress != "" && app.appAddress != "" && appAddress != app.appAddress {
-		app.replyWithError(
-			writer,
-			sdkerrors.Wrapf(
-				ErrInvalidRequestURL,
-				"sender address query parameter does not match the application address: got %s, want %s",
-				appAddress, app.appAddress,
-			),
-		)
-		log.Print("ERROR: application address does not match sender address query parameter")
-		return
-	} else if appAddress == "" {
+	var appAddress string
+
+	if app.appAddress == "" {
```

Comment:
This will never happen as we assign it in the creator. What we want is only to assign in the creator method if we flag the `--self-signing` bool